src/OpenTelemetry/Metrics/CircularBufferBuckets.cs

```diff
@@ -23,7 +23,7 @@ namespace OpenTelemetry.Metrics;
 /// <summary>
 /// A histogram buckets implementation based on circular buffer.
 /// </summary>
-internal sealed class CircularBufferBuckets
+public sealed class CircularBufferBuckets
```

Comment:
@alanwest I don't think we should expose this. It is leaking too much of the implementation.

I'm thinking something more like...

```csharp
public sealed class ExponentialBucketHistogram
{
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public readonly ExponentialBucketHistogramBuckets GetHistogramBuckets() {}
}

public sealed class ExponentialBucketHistogramBuckets // Forgive the name for a second :)
{
    public Enumerator GetEnumerator() => new(this);

    public struct Enumerator
    {
         public HistogramBucket Current { get; private set; } // Maybe we can reuse HistogramBucket here?

         // Not shown implementation of MoveNext which uses internal circular buffer thing
    }
}
```

Reply:
Yes I agree. I also do not want to expose the circular buffer buckets. It was just a temporary shortcut to get something working end-to-end. Using a similar pattern and implementing an enumerator for the buckets was my thought as well. I have a TODO noting this.

https://github.com/open-telemetry/opentelemetry-dotnet/pull/4053/files#diff-7232c9d0a0b2e269249337ed6bd15bcfcce4c4c69bd754e11206e7cc232ee107R304-R313

Regarding reusing HistogramBucket somehow... I don't think this will work. Exponential histograms have two sets of buckets: one for positive values the other for negative. The HistogramBucket class is geared towards modeling a bucket with explicit bounds.